app/src/main/java/ch/epfl/sdp/blindly/utils/UserHelper.kt

```diff
@@ -6,109 +6,123 @@ import android.content.Intent
 import com.google.firebase.ktx.Firebase
-
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Inject
+
+// Module to be installed in Activities
+@Module @InstallIn(SingletonComponent::class)
+class UserHelperModule {
+    @Provides
+    fun provideUserHelper(): UserHelper = UserHelper()
+}
 
 class UserHelper {
-
     companion object {
+        public const val RC_SIGN_IN = 123
+    }
 
-        public fun getSignInIntent(): Intent {
-            // Optionnaly get phone number to set default in login form
-            val phoneProvider = AuthUI.IdpConfig.PhoneBuilder()
-            /*
-            if (ActivityCompat.checkSelfPermission(activity, Manifest.permission.READ_PHONE_NUMBERS) == PackageManager.PERMISSION_GRANTED ){
-                var tMgr: TelephonyManager = activity.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
-                phoneProvider.setDefaultNumber(tMgr.line1Number)
-            }*/
-            val providers = arrayListOf(
-                    AuthUI.IdpConfig.EmailBuilder().setRequireName(false).build(),
-                    phoneProvider.build(),
-                    AuthUI.IdpConfig.GoogleBuilder().build()
-            )
-
-
-            return AuthUI.getInstance()
-                            .createSignInIntentBuilder()
-                            .setAvailableProviders(providers)
-                            //.setLogo(R.drawable.my_great_logo) // Set logo drawable
-                            .setTheme(R.style.Theme_Blindly) // Set theme
-                            /*.setTosAndPrivacyPolicyUrls(
-                    "https://example.com/terms.html",
-                    "https://example.com/privacy.html")*/
-                            .build();
+    fun getSignInIntent(): Intent {
+        // Optionnaly get phone number to set default in login form
+        val phoneProvider = AuthUI.IdpConfig.PhoneBuilder()
+        /*
+        if (ActivityCompat.checkSelfPermission(activity, Manifest.permission.READ_PHONE_NUMBERS) == PackageManager.PERMISSION_GRANTED ){
+            var tMgr: TelephonyManager = activity.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
+            phoneProvider.setDefaultNumber(tMgr.line1Number)
+        }*/
+        val providers = arrayListOf(
+            AuthUI.IdpConfig.EmailBuilder().setRequireName(false).build(),
+            phoneProvider.build(),
+            AuthUI.IdpConfig.GoogleBuilder().build()
+        )
+
+
+        return AuthUI.getInstance()
+            .createSignInIntentBuilder()
+            .setAvailableProviders(providers)
+            //.setLogo(R.drawable.my_great_logo) // Set logo drawable
+            .setTheme(R.style.Theme_Blindly) // Set theme
+            /*.setTosAndPrivacyPolicyUrls(
+    "https://example.com/terms.html",
+    "https://example.com/privacy.html")*/
+            .build();
```

Comment:
Might be a good idea to clean up these comments and keep only what is useful (or explain why it is in comments for now).

Reply:
Well for the comments it is just have these codes available when we will need them, I think adding login by phone could  interset us. Then we will probably add ToS and policy one day.

I admit that I could already add the logo now.